ui/chat/src/ctrl.ts

```diff
@@ -38,6 +38,15 @@ export default class ChatCtrl {
     readonly opts: ChatOpts,
     readonly redraw: Redraw,
   ) {
+    const competitionRegex = new RegExp(
+      `^https:\\/\\/lichess\\.org\\/((tournament)|(swiss))\\/${tempStorage.get('competition.id')}\\/?$`,
+    );
+    const newGameRegex = new RegExp(`^https:\\/\\/lichess\\.org\\/(${tempStorage.get('newGame.id')})\\/?$`);
```

Comment:
ui/chat should know about chatting. It should not know about:

- lichess.org
- https
- tournaments and swiss
- URL formats for competitions and games

If this change requires adding that kind of hacks, then I say it's not worth it.